### PR TITLE
led3: thread type fix

### PR DESCRIPTION
**Pull request by 0xc0170**
Do not use internal rtx type. This will be removed from our includes soon.

### CODE REVIEW

**Comment by 0xc0170** [comment]
The cleanup happens https://github.com/ARMmbed/mbed-os/pull/7364. This fix however do not require it, as the mbed thread storage type is already in the codebase just not used here

**Review by orenc17 — DISMISSED** [review]

**Comment by 0xc0170** [comment]
All green , should be good to go

**Review by alzix — DISMISSED** [review]

File: source/led3.cpp [hunk 1]
```diff
@@ -62,7 +62,7 @@ static void led3_main(const void *)
     SecureAllocator alloc = secure_allocator_create_with_pages(4 * kB, 1 * kB);
     /* Prepare the thread definition structure. */
     osThreadAttr_t thread_attr = {0};
-    mbed_rtos_storage_thread_t thread_def = {0};
```
Comment (0xc0170) [inline source/led3.cpp]: I think we should use this mbed storage instead, so no RTX types

Comment (alzix) [reply source/led3.cpp]: Oren. I was about to send same comment. But Martin was  faster.
This is mbed-os-example-* mbed-os types are most appropriate here.

**Review by alzix — APPROVED** [review]

**Pull request merged**